core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/PartialSuccessHandler.java

```diff
@@ -0,0 +1,145 @@
+package com.microsoft.applicationinsights.channel.concrete.inprocess;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.zip.GZIPInputStream;
+
+import org.apache.http.HttpStatus;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.common.base.Optional;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandler;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandlerArgs;
+import com.microsoft.applicationinsights.internal.channel.common.GzipTelemetrySerializer;
+import com.microsoft.applicationinsights.internal.channel.common.Transmission;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicyManager;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+public class PartialSuccessHandler implements TransmissionHandler {
+
+	public PartialSuccessHandler(TransmissionPolicyManager policy) {
+
+	}
+
+	@Override
+	public void onTransmissionSent(TransmissionHandlerArgs args) {
+		validateTransmissionAndSend(args);
+
+	}
+
+	public boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
+		if (args.getTransmission() != null && args.getTransmissionDispatcher() != null) {
+			switch (args.getResponseCode()) {
+			case HttpStatus.SC_PARTIAL_CONTENT:
+				BackendResponse beR = getBackendResponse(args.getResponseBody());
+				
+				// In case the 206 was false we can break here
+				if(beR != null && (beR.itemsAccepted == beR.itemsReceived))
+				{
+					return false;
+				}
+			
+				ArrayList<String> originalItems = generateOriginalItems(args);
+				
+				// Somehow the amount of items received and the items sent do not match
+				if(beR != null && (originalItems.size() != beR.itemsReceived))
+				{
+					return false;
+				}
+			
+				
+				ArrayList<String> newTransmission = new ArrayList<String>();
+				for (BackendResponse.Error e : beR.errors) {
+					switch (e.statusCode) {
+					case HttpStatus.SC_REQUEST_TIMEOUT:
+					case HttpStatus.SC_INTERNAL_SERVER_ERROR:
+					case HttpStatus.SC_SERVICE_UNAVAILABLE:
+					case 429:
```

Comment:
So we don't handle 429 ?

Reply:
Eh? In the partial success handler any an all error messages are treated the same. In this instance the partial success will try to resend any of these items in the next transmission.

https://github.com/Microsoft/ApplicationInsights-dotnet/blob/0fe5edf099ecdad37990be0c776b300e7fbcff4a/src/ServerTelemetryChannel/Implementation/PartialSuccessTransmissionPolicy.cs#L107-L111

Reply:
Ah got it. So this means if partial message was accepted, then we would retry the whole packet correct?